OsmAnd/src/net/osmand/plus/views/GPXLayer.java

```diff
@@ -326,10 +327,57 @@ private void drawSelectedFilesPoints(Canvas canvas, RotatedTileBox tileBox, List
 	private void drawSelectedFilesSegments(Canvas canvas, RotatedTileBox tileBox,
 			List<SelectedGpxFile> selectedGPXFiles, DrawSettings settings) {
 		for (SelectedGpxFile g : selectedGPXFiles) {
-			List<TrkSegment> points = g.getPointsToDisplay();
-			boolean routePoints = g.isRoutePoints();
-			updatePaints(g.getColor(), routePoints, g.isShowCurrentTrack(), settings, tileBox);
-			drawSegments(canvas, tileBox, points);
+			List<TrkSegment> segments = g.getPointsToDisplay();
+			for (TrkSegment ts : segments) {
+
+				// TODO: Select/install renderables via UI or external process!  (See comments below)
+
```

Comment:
This is quite a long text and doesn't look to be at the right place in sense I see the mixture of what is going on in development how it could work or how it should not work. It might not be the right place to have here.
I think if somebody will need to dive to that code he just need to understand exactly what's going on he doesn't need to understand the plans of future implementation. 
Personally I'm against comments because I think it is always possible to refactor the code that it doesn't have comments. For example, TrkSegment has one or more 'Renderables' - it could be clear from the code itself (1 to many relations). What is Renderable is much better to describe in the Renderable class header itself. Second part Renderables are processed asynchronously is kind of obvious because we see AsyncTask.
"Note" is a good to know for us in pull request but not in the code may be.